PLUGIN_MIGRATION.md

```diff
@@ -0,0 +1,60 @@
+# Plugin Migration
+
+*Note*: This document is evolving and is in draft state.
+
+A goal of the Extensions SDK is to ease developer migration of plugins to extensions.  Supporting this goal:
+ - Plugin interfaces and their extension points are being migrated to equivalent Extension interfaces
+ - Same-named methods and arguments are used whenever possible, with differences highlighted in this migration guide
+ - Wrapper classes are provided when possible to retain existing code logic
+
+Migration of the [Anomaly Detection Plugin](https://github.com/opensearch-project/anomaly-detection) to an Extension is in progress to both identify and refine migration challenges and provide examples to other plugin developers.
+
+## Implement Extension Interfaces and Extension Points
+
+Change the implementing class `FooPlugin` to `FooExtension` and implement the `Extension` interface.
+ - Implement other corresponding Extension interfaces, for example `ActionPlugin` interface would be `ActionExtension`.
+ - Change extension point implementation, if necessary, to conform to new types. For example, `RestHandler` classes become `ExtensionRestHandler` classes.  Consider extending base classes such as `BaseExtensionRestHandler` to provide additional convenience methods.
+
+## Use Wrapper Classes
+
+### Replace ClusterService with SDKClusterService
+
+ - `clusterService.getClusterSettings().addSettingsUpdateConsumer()` works as is, but has a method taking a map parameter can do multiple updates more efficiently.
```

Comment:
The wording of the second half of this line is a little confusing

Reply:
+1